pkg/apis/machine/v1alpha1/types.go

```diff
@@ -850,6 +850,18 @@ type AWSEbsBlockDeviceSpec struct {
 	// it is not used in requests to create gp2, st1, sc1, or standard volumes.
 	Iops int64 `json:"iops,omitempty"`
 
+	// Identifier (key ID, key alias, ID ARN, or alias ARN) for a customer managed
+	// CMK under which the EBS volume is encrypted.
+	//
+	// This parameter is only supported on BlockDeviceMapping objects called by
+	// RunInstances (https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_RunInstances.html),
+	// RequestSpotFleet (https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_RequestSpotFleet.html),
+	// and RequestSpotInstances (https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_RequestSpotInstances.html).
+	KmsKeyID *string `json:"kmsKeyID,omitempty"`
```

Comment:
Why is this required? And if required, we will need some code changes also to accommodate the change right?

Reply:
This field is not required for the snapshot support. 
This will potentially be needed soon for the encryption of the worker-volumes. The unimplemented field doesn't seem to be hurting the flow, as the related implementation can always be added later.

Reply:
Okay.